### DISCUSSION TITLE
Christmas Wishes

### DISCUSSION THREAD

**Discussion by emuskardin** [DES-Lab/AALpy#57] [Ideas]
Dear @mtappler , @zwergziege , @icezyclon , @apferscher , @aichernig , and others,

If you have any wishes you want to see in AALpy next year, write them down here so we can have a discussion, or just gather ideas.
On my wish list:
- examine KV for non-deterministic/stochastic systems
- work/extend Wiki and Examples.py (Tips and Tricks section for advanced use for example)
- passive context-free grammar learning (deterministic and "visible" alphabet (divided in push/pull/internal symbols))

![](https://i.pinimg.com/originals/63/7b/f9/637bf9819dace179e4c2a71fea0d0356.png).

*Happy Holidays*

**Comment by icezyclon** [comment]
Merry Christmas!

Well, if I were to write a wish list it would probably include an _implementation of HW_, that would have come in handy already. 
Otherwise, _converting between machines_ might be useful? I only remember needing to convert from Mealy to Moore (which I did at that time by re-learning the machine, but I'm sure there is a separate algorithm for efficient conversion).

Wishing you and AALpy a happy new year!🎄🎆

**Reply by icezyclon** [reply]
One more point, *please* add top-level import statements! You can keep the package separation but let me write `from aalpy import MealyMachine, run_Lstar, ...`.
This would make it SO much easier to find stuff and you can separate imports if you want to. This would be great!

**Reply by emuskardin** [reply]
Added. You can pull the latest master branch to test

**Reply by icezyclon** [reply]
I pushed a fix for a mistake and made the imports explicit. Opened a pull request :+1:
